spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -460,6 +462,37 @@ def not_ok?; false; end
             end
           end
         end
+
+        specify 'memoized blocks warns against deadlock' do
+          describe_successfully do
+            let!(:klass) do
+              Class.new do
+                def bar
+                  return 1
+                end
+              end
+            end
+            # This is accesssed inside of the subject in a different thread,
+            # causing deadlock.
+            let(:value) { 5 }
+
+            before do
+              allow_any_instance_of(klass).to receive(:bar) do
+                value
+              end
+            end
+
+            subject do
+              Thread.new do
+                klass.new.bar
+              end.value
+            end
+
+            example do
+              expect { subject }.to raise_error(/deadlock/)
+            end
+          end
+        end
```

Comment:
I think this accomplishes the same thing, and requires a lot less analysis of the test in order to understand it.

```ruby
require 'rspec/autorun'
RSpec.describe do
  let(:lock) do
    Thread.new { lock }.join
  end
  example do
    expect { lock }.to raise_error(/deadlock/)
  end
end

# >> .
# >> 
# >> Finished in 0.10371 seconds (files took 0.20367 seconds to load)
# >> 1 example, 0 failures
# >> 
```